lib/dts-lib.robot

```diff
@@ -0,0 +1,130 @@
+*** Settings ***
+Library     ../robot-venv/lib/python3.11/site-packages/robot/libraries/Collections.py
+
+
+*** Keywords ***
+Enable Network Boot
+    Power On
+    ${setup_menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${dasharo_menu}=    Enter Dasharo System Features    ${setup_menu}
+    ${nw_menu}=    Enter Dasharo Submenu    ${dasharo_menu}    Networking Options
+    Set Option State    ${nw_menu}    Enable network boot    ${TRUE}
+    Save Changes And Reset    2    4
+
+Boot Dasharo Tools Suite
+    [Documentation]    Keyword allows to boot Dasharo Tools Suite. Takes the
+    ...    boot method (from USB or from iPXE) as parameter.
+    [Arguments]    ${dts_booting_method}
+    ${boot_menu}=    Enter Boot Menu Tianocore And Return Construction
+    IF    '${dts_booting_method}'=='USB'
+        Enter Submenu From Snapshot    ${boot_menu}    USB SanDisk 3.2Gen1
+    ELSE IF    '${dts_booting_method}'=='USB_emulated'
+        Enter Submenu From Snapshot    ${boot_menu}    PiKVM Composite KVM Device
+        Sleep    1s
+        Press Key N Times    1    ${ENTER}
+    ELSE IF    '${dts_booting_method}'=='iPXE'
+        Enter Submenu From Snapshot    ${boot_menu}    iPXE Network Boot
+        ${ipxe_menu}=    Get Submenu Construction    iPXE Shell
+        Enter Submenu From Snapshot    ${ipxe_menu}    Dasharo Tools Suite
+    ELSE
+        FAIL    Unknown or improper connection method: ${dts_booting_method}
+    END
+    Read From Terminal Until    Enter an option:
+
+Check HCL Report Creation
+    [Documentation]    Keyword allows to check if the Dasharo Tools Suite
+    ...    option for creating HCL report works correctly.
+    Enter Shell In DTS
+    Set Global Variable    ${DUT_CONNECTION_METHOD}    SSH
+    Execute Command In Terminal    cd /
+    ${logs}=    Execute Command In Terminal
+    ...    command=/usr/bin/env DEPLOY_REPORT=false SEND_LOGS=true /usr/sbin/dasharo-hcl-report
+    ...    timeout=210s
+    Should Contain    ${logs}    Thank you
+    Should Contain    ${logs}    exited without errors
+    Should Contain    ${logs}    send completed
+
+Enter Shell In DTS
+    [Documentation]    Keyword allows to drop to Shell in the Dasharo Tools
+    ...    Suite.
+    Write Into Terminal    8
+    Login To Linux Via SSH Without Password    root    root@DasharoToolsSuite:
+
+Run EC Transition
+    [Documentation]    Keyword allows to run EC Transition procedure in the
+    ...    Dasharo Tools Suite.
+    Write Into Terminal    6
+    Read From Trminal Until    Enter an option:
+    Write Into Terminal    1
+    ${output}=    Read From Terminal Until    shut down
+    Should Contain X Times    ${output}    VERIFIED    2
+    Sleep    10s
+
+Check Power Off In DTS
+    [Documentation]    Keyword allows to check if the Dasharo Tools Suite
+    ...    option for power off the DUT works correctly..
+    Sleep    5s
+    ${output}=    Read From Terminal
+    Length Should Be    ${output}    1
+
+Flash Firmware In DTS
```

Comment:
This is much more complex, and command may be different depending on the platform.

Moreover, we have 2 locks in setup menu to take care of, that can make flashing to fail.

Reply:
Added a [keyword](https://github.com/Dasharo/open-source-firmware-validation/pull/93/commits/610597bc34b1431fb44fc5599c0c826cd14b0e98#diff-6ad7c5fb97cb1099c7ecd42f15dde3caf06d38b9fe5244afb96394454b6d6bfaR537) to menus for disabling the locks, TODO meeting different flashing command requirements

Reply:
This will be improved by @Psotas , we have something simlar already: https://github.com/Dasharo/open-source-firmware-validation/blob/main/keys-and-keywords/flashrom.robot

Please link issue here when created